CONTRIBUTING.md

```diff
@@ -1,39 +1,65 @@
-# Contributing
+# How to Contribute
 
-## How to Contribute
+1. Report bugs as github issues. We don't use a template, but try to provide
+   some sensible information such as what what happened, what you expected to
+   happen and steps to reproduce. If applicable try with default configuration.
+   If you are able to, try to do some debugging (guidelines below).
 
-## Debugging
+2. Submit patches as github pull-requests. If the patch introduces significant
+   changes or new features, discuss on IRC or via gh issue first to avoid
+   wasting time. Also consult the scope document to ensure it's in keeping.
```

Comment:
"wasting time" -> "wasting your time"? Not sure if the first implies the second. Not a native english speaker so this might be totally fine.

> Also consult the scope document to ensure it's in keeping.

Maybe we could link "scope document"?

> to ensure it's in keeping.

"keeping" == Category `A`, `B+` or `B`?

Could

Reply:
"wasting time" could probably mean "your time", "our time", or both. Which might all be true in some cases :)

Reply:
"wasting time" relates to everyone's time. I've toned it down a bit though to sound more friendly (which is how it is intended)

Have added link to scope document.

"In keeping" = "consistent" / "doesn't deviate". I have refactored the sentence to remove this ambiguous term.

Reply:
Right but "wasting our time" sounds a bit arrogant, "wasting your time" is sensible. So what I was referring to was to not give the wrong impression of "RTFM to not waste our time".

Reply:
Yes - I'll leave it out to avoid being offensive to anyone... however true it might all be :smile: